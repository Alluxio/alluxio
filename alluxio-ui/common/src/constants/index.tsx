export * from './types/INavigationData';
export * from './types/IFileBlockInfo';
export * from './types/IFileInfo';
