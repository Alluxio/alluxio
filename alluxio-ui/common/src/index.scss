/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

@import './constants/scss/themeVariables';
@import '~bootstrap/scss/bootstrap';
@import '~bootstrap/scss/bootstrap-grid';
@import '~bootstrap/scss/bootstrap-reboot';
@import './constants/scss/themeCustomization';
@import '~source-sans-pro/source-sans-pro.css';

body {
  margin: 0;
  padding: 0;
  font-family: 'Source Sans Pro', sans-serif;
  font-size: 1em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-shadow: rgba(0,0,0,.01) 0 0 1px;
  text-shadow: rgba(0,0,0,.01) 0 0 1px;
}
