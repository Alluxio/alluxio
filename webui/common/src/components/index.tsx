/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

export * from './HOCs/withLoadingMessage/withLoadingMessage';
export * from './HOCs/withErrors/withErrors';
export * from './HOCs/withTextAreaResize/withTextAreaResize';
export * from './HOCs/withFetchDataFromPath/withFetchDataFromPath';
export * from './HOCs/withFetchData/withFetchData';
export * from './HOCs/withFluidContainer/withFluidContainer';
export * from './Footer/Footer';
export * from './Header/Header';
export * from './LoadingMessage/LoadingMessage';
export * from './Logs/Logs';
export * from './FileView/FileView';
export * from './Paginator/Paginator';
export * from './LineGraph/LineGraph';
