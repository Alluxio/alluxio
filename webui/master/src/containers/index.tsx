/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

export {default as App} from './App/App';

export {default as Browse} from './pages/Browse/Browse';
export {default as Configuration} from './pages/Configuration/Configuration';
export {default as Data} from './pages/Data/Data';
export {default as Overview} from './pages/Overview/Overview';
export {default as MasterLogs} from './pages/MasterLogs/MasterLogs';
export {default as Workers} from './pages/Workers/Workers';
export {default as Metrics} from './pages/Metrics/Metrics';
