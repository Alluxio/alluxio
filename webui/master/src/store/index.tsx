/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {connectRouter, RouterState} from 'connected-react-router';
import {History} from 'history';
import {combineReducers} from 'redux';
import {all, fork} from 'redux-saga/effects';

import {initialRefreshState, refreshReducer} from '@alluxio/common-ui/src/store/refresh/reducer';
import {IRefreshState} from '@alluxio/common-ui/src/store/refresh/types';
import {initialLogsState, logsReducer} from '@alluxio/common-ui/src/store/logs/reducer';
import {logsSaga} from '@alluxio/common-ui/src/store/logs/sagas';
import {ILogsState} from '@alluxio/common-ui/src/store/logs/types';
import {browseReducer, initialBrowseState} from './browse/reducer';
import {browseSaga} from './browse/sagas';
import {IBrowseState} from './browse/types';
import {configReducer, initialConfigState} from './config/reducer';
import {configSaga} from './config/sagas';
import {IConfigState} from './config/types';
import {dataReducer, initialDataState} from './data/reducer';
import {dataSaga} from './data/sagas';
import {IDataState} from './data/types';
import {initialInitState, initReducer} from './init/reducer';
import {initSaga} from './init/sagas';
import {IInitState} from './init/types';
import {initialMetricsState, metricsReducer} from './metrics/reducer';
import {metricsSaga} from './metrics/sagas';
import {IMetricsState} from './metrics/types';
import {initialOverviewState, overviewReducer} from './overview/reducer';
import {overviewSaga} from './overview/sagas';
import {IOverviewState} from './overview/types';
import {initialWorkersState, workersReducer} from './workers/reducer';
import {workersSaga} from './workers/sagas';
import {IWorkersState} from './workers/types';

export interface IApplicationState {
  browse: IBrowseState;
  config: IConfigState;
  data: IDataState;
  init: IInitState;
  logs: ILogsState;
  metrics: IMetricsState;
  overview: IOverviewState;
  refresh: IRefreshState;
  router?: RouterState;
  workers: IWorkersState;
}

export const rootReducer = (history: History) => combineReducers<IApplicationState>({
  browse: browseReducer,
  config: configReducer,
  data: dataReducer,
  init: initReducer,
  logs: logsReducer,
  metrics: metricsReducer,
  overview: overviewReducer,
  refresh: refreshReducer,
  router: connectRouter(history),
  workers: workersReducer
});

export const rootSaga = function* () {
  yield all([
    fork(browseSaga),
    fork(configSaga),
    fork(dataSaga),
    fork(initSaga),
    fork(logsSaga),
    fork(metricsSaga),
    fork(overviewSaga),
    fork(workersSaga)
  ]);
};

export const initialState: IApplicationState = {
  browse: initialBrowseState,
  config: initialConfigState,
  data: initialDataState,
  init: initialInitState,
  logs: initialLogsState,
  metrics: initialMetricsState,
  overview: initialOverviewState,
  refresh: initialRefreshState,
  workers: initialWorkersState
};
