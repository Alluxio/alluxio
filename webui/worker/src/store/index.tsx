/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {connectRouter, RouterState} from 'connected-react-router';
import {History} from 'history';
import {combineReducers} from 'redux';
import {all, fork} from 'redux-saga/effects';

import {initialRefreshState, refreshReducer} from '@alluxio/common-ui/src/store/refresh/reducer';
import {IRefreshState} from '@alluxio/common-ui/src/store/refresh/types';
import {initialLogsState, logsReducer} from '@alluxio/common-ui/src/store/logs/reducer';
import {logsSaga} from '@alluxio/common-ui/src/store/logs/sagas';
import {ILogsState} from '@alluxio/common-ui/src/store/logs/types';
import {blockInfoReducer, initialBlockInfoState} from './blockInfo/reducer';
import {blockInfoSaga} from './blockInfo/sagas';
import {IBlockInfoState} from './blockInfo/types';
import {initialMetricsState, metricsReducer} from './metrics/reducer';
import {metricsSaga} from './metrics/sagas';
import {IMetricsState} from './metrics/types';
import {initialOverviewState, overviewReducer} from './overview/reducer';
import {overviewSaga} from './overview/sagas';
import {IOverviewState} from './overview/types';
import {IInitState} from './init/types';
import {initialInitState, initReducer} from './init/reducer';
import {initSaga} from './init/sagas';

export interface IApplicationState {
  blockInfo: IBlockInfoState;
  init: IInitState;
  logs: ILogsState;
  metrics: IMetricsState;
  overview: IOverviewState;
  refresh: IRefreshState;
  router?: RouterState;
}

export const rootReducer = (history: History) => combineReducers<IApplicationState>({
  blockInfo: blockInfoReducer,
  init: initReducer,
  logs: logsReducer,
  metrics: metricsReducer,
  overview: overviewReducer,
  refresh: refreshReducer,
  router: connectRouter(history)
});

export const rootSaga = function* () {
  yield all([
    fork(blockInfoSaga),
    fork(initSaga),
    fork(logsSaga),
    fork(metricsSaga),
    fork(overviewSaga)
  ]);
};

export const initialState: IApplicationState = {
  blockInfo: initialBlockInfoState,
  init: initialInitState,
  logs: initialLogsState,
  metrics: initialMetricsState,
  overview: initialOverviewState,
  refresh: initialRefreshState
};
