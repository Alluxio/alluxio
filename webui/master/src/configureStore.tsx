/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {routerMiddleware} from 'connected-react-router';
import {History} from 'history';
import {applyMiddleware, createStore, Store} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import {IApplicationState, rootReducer, rootSaga} from './store';

export default function configureStore(history: History, initialState: IApplicationState): Store<IApplicationState> {
  const composeEnhancers = composeWithDevTools({});
  const sagaMiddleware = createSagaMiddleware();

  const store = createStore(
    rootReducer(history),
    initialState,
    composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware))
  );

  sagaMiddleware.run(rootSaga);
  return store;
}
