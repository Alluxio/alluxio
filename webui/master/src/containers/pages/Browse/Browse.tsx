/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {faFile, faFolder} from '@fortawesome/free-regular-svg-icons'
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
import {History, LocationState} from 'history';
import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {Button, Form, FormGroup, Input, Label, Table} from 'reactstrap';
import {compose, Dispatch} from 'redux';

import {
  FileView, withErrors, withFetchDataFromPath, withFluidContainer, withLoadingMessage,
  withTextAreaResize,
  Paginator
} from '@alluxio/common-ui/src/components';
import {IAlertErrors, IFileBlockInfo, IFileInfo, IRequest, ICommonState} from '@alluxio/common-ui/src/constants';
import {
  createAlertErrors,
  disableFormSubmit,
  renderFileNameLink
} from '@alluxio/common-ui/src/utilities';
import {IApplicationState} from '../../../store';
import {fetchRequest} from '../../../store/browse/actions';
import {IBrowse} from '../../../store/browse/types';
import {IInit} from '../../../store/init/types';

import './Browse.css';
import {routePaths} from "../../../constants";

export interface IPropsFromState extends ICommonState {
  browseData: IBrowse,
  initData: IInit
}

interface IPropsFromDispatch {
  fetchRequest: typeof fetchRequest;
}

interface IBrowseProps {
  history: History<LocationState>;
  location: {search: string};
  queryStringSuffix: string;
  request: IRequest;
  textAreaHeight: number;
  upateRequestParameter: (reqParam: string, value: string | undefined) => void;
}

export type AllProps = IPropsFromState & IBrowseProps & IPropsFromDispatch;

export class BrowsePresenter extends React.Component<AllProps> {
  public render() {
    const {browseData, initData, queryStringSuffix} = this.props;

    return (
      <div className="col-12">
        {!browseData.currentDirectory.isDirectory
            ? this.renderFileView(initData, browseData, queryStringSuffix)
            : this.renderDirectoryListing(initData, browseData)}
      </div>
    );
  }

  private renderFileView(initData: IInit, browseData: IBrowse, queryStringSuffix: string) {
    const {request: {offset, end, path}, history, textAreaHeight} = this.props;
    const offsetInputHandler = this.createInputChangeHandler('offset').bind(this);
    const beginInputHandler = this.createButtonHandler('end', undefined).bind(this);
    const endInputHandler = this.createButtonHandler('end', '1').bind(this);
    return (
        <React.Fragment>
          <FileView allowDownload={true} beginInputHandler={beginInputHandler} end={end} endInputHandler={endInputHandler}
                    offset={offset} offsetInputHandler={offsetInputHandler} path={path}
                    queryStringPrefix={routePaths.browse} queryStringSuffix={queryStringSuffix} textAreaHeight={textAreaHeight}
                    viewData={browseData} history={history} proxyDownloadApiUrl={initData.proxyDownloadFileApiUrl}/>
          <hr/>
          <h6>Detailed blocks information (block capacity is {browseData.blockSizeBytes} Bytes):</h6>
          <Table hover={true}>
            <thead>
            <tr>
              <th>ID</th>
              <th>Size (Byte)</th>
              <th>In {browseData.highestTierAlias}</th>
              <th>Locations</th>
            </tr>
            </thead>
            <tbody>
            {browseData.fileBlocks.map((fileBlock: IFileBlockInfo) => (
              <tr key={fileBlock.id}>
                <td>{fileBlock.id}</td>
                <td>{fileBlock.blockLength}</td>
                <td>
                  {fileBlock.isInHighestTier ? 'YES' : 'NO'}
                </td>
                <td>
                  {fileBlock.locations.map((location: string) => (
                    <div key={location}>{location}</div>
                  ))}
                </td>
              </tr>
            ))}
            </tbody>
          </Table>
        </React.Fragment>
    );
  }

  private renderDirectoryListing(initData: IInit, browseData: IBrowse) {
    const {request: {limit, offset, path}, history} = this.props;
    const fileInfos = browseData.fileInfos;
    const pathInputHandler = this.createInputChangeHandler('path').bind(this);
    return (
      <React.Fragment>
        <Form className="mb-3 browse-directory-form" id="browseDirectoryForm" inline={true}
              onSubmit={disableFormSubmit}>
          <FormGroup className="mb-2 mr-sm-2">
            <Button tag={Link} to={`/browse?path=/`} color="secondary"
                    outline={true}>Root</Button>
          </FormGroup>
          <FormGroup className="mb-2 mr-sm-2">
            <Label for="browsePath" className="mr-sm-2">Path</Label>
            <Input type="text" id="browsePath" placeholder="Enter a Path" value={path || '/'}
                   onChange={pathInputHandler}
                   onKeyUp={this.createInputEnterHandler(history, () =>
                   `/browse?path=${path}`)}/>
          </FormGroup>
          <FormGroup className="mb-2 mr-sm-2">
            <Button tag={Link} to={`/browse?path=${encodeURIComponent(path || '/')}`}
            color="secondary">Go</Button>
          </FormGroup>
        </Form>
        <Table hover={true}>
          <thead>
          <tr>
            <th/>
            {/* Icon placeholder */}
            <th>File Name</th>
            <th>Size</th>
            <th>Block Size</th>
            <th>In-Alluxio</th>
            {browseData.showPermissions && (
              <React.Fragment>
                <th>Mode</th>
                <th>Owner</th>
                <th>Group</th>
              </React.Fragment>
            )}
            <th>Persistence State</th>
            <th>Pin</th>
            <th>Creation Time</th>
            <th>Modification Time</th>
            {initData.debug && (
              <React.Fragment>
                <th>[D]DepID</th>
                <th>[D]INumber</th>
                <th>[D]UnderfsPath</th>
                <th>[D]File Locations</th>
              </React.Fragment>
            )}
          </tr>
          </thead>
          <tbody>
          {fileInfos && fileInfos.map((fileInfo: IFileInfo) => (
            <tr key={fileInfo.absolutePath}>
              <td><FontAwesomeIcon icon={fileInfo.isDirectory ? faFolder : faFile}/></td>
              <td>
                {renderFileNameLink(fileInfo.absolutePath, `/browse?path=`)}
              </td>
              <td>{fileInfo.size}</td>
              <td>{fileInfo.blockSizeBytes}</td>
              <td>{fileInfo.inAlluxioPercentage}%</td>
              {browseData.showPermissions && (
                <React.Fragment>
                  <td>
                    <pre className="mb-0"><code>{fileInfo.mode}</code></pre>
                  </td>
                  <td>{fileInfo.owner}</td>
                  <td>{fileInfo.group}</td>
                </React.Fragment>
              )}
              <td>{fileInfo.persistenceState}</td>
              <td>{fileInfo.pinned ? 'YES' : 'NO'}</td>
              <td>{fileInfo.creationTime}</td>
              <td>{fileInfo.modificationTime}</td>
              {initData.debug && (
                <React.Fragment>
                  <td>{fileInfo.id}</td>
                  <td>
                    {fileInfo.fileLocations.map((location: string) => <div key={location}>location</div>)}
                  </td>
                  <td>{fileInfo.absolutePath}</td>
                  <td>
                    {fileInfo.fileLocations.map((location: string) => (
                      <div key={location}>{location}</div>
                    ))}
                  </td>
                </React.Fragment>
              )}
            </tr>
          ))}
          </tbody>
        </Table>
        <Paginator baseUrl={routePaths.browse} path={path} total={browseData.ntotalFile} offset={offset} limit={limit}/>
      </React.Fragment>
    )
  }

  private createInputChangeHandler(reqParam: string) {
    return (event: React.ChangeEvent<HTMLInputElement>) => {
      this.props.upateRequestParameter(reqParam, event.target.value);
    };
  }

  private createButtonHandler(reqParam: string, value: string | undefined) {
    return (event: React.MouseEvent<HTMLButtonElement>) => {
      this.props.upateRequestParameter(reqParam, value);
    };
  }

  private createInputEnterHandler(history: History<LocationState>, stateValueCallback: (value: string) => string | undefined) {
    return (event: React.KeyboardEvent<HTMLInputElement>) => {
      const value = event.key;
      if (event.key === 'Enter') {
        const newPath = stateValueCallback(value);
        if (newPath) {
          if (history) {
            history.push(newPath);
          }
        }
      }
    };
  }
}

const mapStateToProps = ({browse, init, refresh}: IApplicationState): IPropsFromState => {
  const errors: IAlertErrors = createAlertErrors(
      init.errors !== undefined || browse.errors !== undefined,
      [
        browse.data.accessControlException,
        browse.data.fatalError,
        browse.data.fileDoesNotExistException,
        browse.data.invalidPathError,
        browse.data.invalidPathException
      ]
  );

  return {
    browseData: browse.data,
    initData: init.data,
    refresh: refresh.data,
    loading: browse.loading || init.loading,
    class: 'browse-page',
    errors: errors
  };
};

const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchRequest: (request: IRequest) => dispatch(fetchRequest(request))
});

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withFetchDataFromPath,
  withErrors,
  withLoadingMessage,
  withTextAreaResize,
  withFluidContainer,
)(BrowsePresenter) as React.Component;
