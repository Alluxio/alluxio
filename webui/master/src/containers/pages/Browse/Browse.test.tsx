/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {configure, mount, ReactWrapper, shallow, ShallowWrapper} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {createBrowserHistory, History, LocationState} from 'history';
import React from 'react';
import {Provider} from 'react-redux';
import {Store} from 'redux';
import sinon, {SinonSpy} from 'sinon';

import configureStore from '../../../configureStore'
import {initialState, IApplicationState} from '../../../store';
import ConnectedApp from '../../App/App'
import Browse, {AllProps, BrowsePresenter} from './Browse';
import {routePaths} from "../../../constants";
import {createAlertErrors} from "@alluxio/common-ui/src/utilities";

configure({adapter: new Adapter()});

describe('Browse', () => {
  let history: History<LocationState>;
  let store: Store<IApplicationState>;
  let props: AllProps;

  beforeAll(() => {
    history = createBrowserHistory({keyLength: 0});
    history.push(routePaths.browse);
    store = configureStore(history, initialState);
    props = {
      location: {search: ''},
      fetchRequest: sinon.spy(),
      history: history,
      browseData: initialState.browse.data,
      initData: initialState.init.data,
      refresh: initialState.refresh.data,
      queryStringSuffix: '',
      class: '',
      errors: createAlertErrors(false),
      loading: false,
      textAreaHeight: 0,
      request: {},
      upateRequestParameter: sinon.spy()
    };
  });

  afterEach(() => {
    sinon.restore();
  });

  describe('Shallow component', () => {
    let shallowWrapper: ShallowWrapper;

    beforeAll(() => {
      shallowWrapper = shallow(<BrowsePresenter {...props} />);
    });

    it('Renders without crashing', () => {
      expect(shallowWrapper.length).toEqual(1);
    });

    it('Contains a div with class col-12', () => {
      expect(shallowWrapper.find('.col-12').length).toEqual(1);
    });

    describe('Renders Directory Listing', () => {
      it('Matches snapshot with Table listing', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });

    describe('Renders FileView', () => {
      beforeAll(() => {
        const data = {...props.browseData};
        data.currentDirectory.isDirectory = false;
        shallowWrapper.setProps({browseData: data});
      });

      it('Matches snapshot with File', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });
  });
});
