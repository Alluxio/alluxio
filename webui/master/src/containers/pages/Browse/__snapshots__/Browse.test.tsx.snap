// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browse Shallow component Renders Directory Listing Matches snapshot with Table listing 1`] = `
<div
  className="col-12"
>
  <Form
    className="mb-3 browse-directory-form"
    id="browseDirectoryForm"
    inline={true}
    onSubmit={[Function]}
    tag="form"
  >
    <FormGroup
      className="mb-2 mr-sm-2"
      tag="div"
    >
      <Button
        color="secondary"
        outline={true}
        tag={[Function]}
        to="/browse?path=/"
      >
        Root
      </Button>
    </FormGroup>
    <FormGroup
      className="mb-2 mr-sm-2"
      tag="div"
    >
      <Label
        className="mr-sm-2"
        for="browsePath"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Path
      </Label>
      <Input
        id="browsePath"
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="Enter a Path"
        type="text"
        value="/"
      />
    </FormGroup>
    <FormGroup
      className="mb-2 mr-sm-2"
      tag="div"
    >
      <Button
        color="secondary"
        tag={[Function]}
        to="/browse?path=%2F"
      >
        Go
      </Button>
    </FormGroup>
  </Form>
  <Table
    hover={true}
    responsiveTag="div"
    tag="table"
  >
    <thead>
      <tr>
        <th />
        <th>
          File Name
        </th>
        <th>
          Size
        </th>
        <th>
          Block Size
        </th>
        <th>
          In-Alluxio
        </th>
        <th>
          Mode
        </th>
        <th>
          Owner
        </th>
        <th>
          Group
        </th>
        <th>
          Persistence State
        </th>
        <th>
          Pin
        </th>
        <th>
          Creation Time
        </th>
        <th>
          Modification Time
        </th>
      </tr>
    </thead>
    <tbody />
  </Table>
  <Paginator
    baseUrl="/browse"
    total={0}
  />
</div>
`;

exports[`Browse Shallow component Renders FileView Matches snapshot with File 1`] = `
<div
  className="col-12"
>
  <FileView
    allowDownload={true}
    beginInputHandler={[Function]}
    endInputHandler={[Function]}
    history={
      Object {
        "action": "PUSH",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/browse",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    offsetInputHandler={[Function]}
    proxyDownloadApiUrl={
      Object {
        "prefix": "http://192.168.1.6:39999/api/v1/paths/",
        "suffix": "/download-file/",
      }
    }
    queryStringPrefix="/browse"
    queryStringSuffix=""
    textAreaHeight={0}
    viewData={
      Object {
        "accessControlException": "",
        "blockSizeBytes": "",
        "currentDirectory": Object {
          "absolutePath": "",
          "blockSizeBytes": "",
          "blocksOnTier": Object {},
          "creationTime": "",
          "fileLocations": Array [],
          "group": "",
          "id": 0,
          "inAlluxio": false,
          "inAlluxioPercentage": 0,
          "isDirectory": false,
          "mode": "",
          "modificationTime": "",
          "name": "",
          "owner": "",
          "persistenceState": "",
          "pinned": false,
          "size": "",
        },
        "currentPath": "",
        "debug": false,
        "fatalError": "",
        "fileBlocks": Array [],
        "fileData": "",
        "fileDoesNotExistException": "",
        "fileInfos": Array [],
        "highestTierAlias": "",
        "invalidPathError": "",
        "invalidPathException": "",
        "masterNodeAddress": "",
        "ntotalFile": 0,
        "pathInfos": Array [],
        "showPermissions": true,
        "viewingOffset": 0,
      }
    }
  />
  <hr />
  <h6>
    Detailed blocks information (block capacity is 
     Bytes):
  </h6>
  <Table
    hover={true}
    responsiveTag="div"
    tag="table"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Size (Byte)
        </th>
        <th>
          In 
        </th>
        <th>
          Locations
        </th>
      </tr>
    </thead>
    <tbody />
  </Table>
</div>
`;
